test/cpp/api/sequential.cpp

```diff
@@ -64,6 +65,81 @@ TEST_F(SequentialTest, ConstructsFromModuleHolder) {
   ASSERT_EQ(sequential->size(), 3);
 }
 
+TEST_F(SequentialTest, ConstructsFromSharedPointerWithName) {
+  struct M : torch::nn::Module {
```

Comment:
Why does this test duplicate so much code, won't a single definition of this M struct work?